docs/examples/llama/llama_model.py

```diff
@@ -0,0 +1,316 @@
+            # Output 2, 3 are None
+            return model_output[0], model_output[1]
+
+    def generate_example_inputs(batch: int, seq: int, vocab_size: int):
+        input_ids = ids_tensor([batch, seq], vocab_size)
+        input_mask = torch.tril(torch.ones(batch, seq, dtype=torch.float32))
+        assert input_mask.dtype == torch.float32
+        return input_ids, input_mask
+
+    example_args_collection = []
+    for b, s in input_dims:
+        example_args_collection.append(generate_example_inputs(b, s, vocab_size))
+
+    return LlamaModelWrapper(config), example_args_collection
+
+
+def display_model_stats(model: onnx.ModelProto):
+    """Displays the number of nodes, number of inputs, number of outputs and number of initializers."""
+    print(f"number of nodes: {len(model.graph.node)}")
+    unique_nodes = collections.Counter([node.op_type for node in model.graph.node])
+    for k, v in unique_nodes.items():
+        print(f"  {k}: {v}")
+    print(f"number of inputs: {len(model.graph.input)}")
+    print(f"number of outputs: {len(model.graph.output)}")
+    print(f"number of initializers: {len(model.graph.initializer)}")
+    print(f"number of functions: {len(model.functions)}")
+
+
+def time_ort_model(model, name: str, example_args, run_count=100):
+    session = onnxruntime.InferenceSession(model.SerializeToString())
+    input_data = [x.numpy() for x in example_args]
+    start = time.time()
+    ort_input = dict(zip((input.name for input in session.get_inputs()), input_data))
+    for _ in range(run_count):
+        session.run(None, ort_input)
+    end = time.time()
+    print(f"ORT Time {name}:", (end - start) / run_count, "average over", run_count, "runs")
+
+
+def export():
+    model, example_args_collection = get_llama_model(num_hidden_layers=10)
+
+    exported = torch.export.export(model, example_args_collection[0])
+    print("===exported fx graph===")
+    print(exported)
+    print("FX Node count:", len(exported.graph.nodes))
+    exported_onnx = torch.onnx.dynamo_export(model, *example_args_collection[0]).model_proto
+    print("===exported_onnx===")
+    display_model_stats(exported_onnx)
+    inlined_exported_onnx = onnx.inliner.inline_local_functions(exported_onnx)
+    print("===inlined_exported_onnx===")
+    display_model_stats(inlined_exported_onnx)
+
+    _flags.EXPERIMENTAL_PREFER_TRACING = True
+
+    exported_eager_onnx = torch.onnx.dynamo_export(
+        model, *example_args_collection[0]
+    ).model_proto
+    print("===exported_eager_onnx===")
+    display_model_stats(exported_eager_onnx)
+    inlined_eager_exported_onnx = onnx.inliner.inline_local_functions(exported_eager_onnx)
+    print("===inlined_eager_exported_onnx===")
+    display_model_stats(inlined_eager_exported_onnx)
+    # print(onnx.printer.to_text(inlined_eager_exported_onnx))
+    onnx.save(inlined_eager_exported_onnx, "inlined_eager_exported_onnx.onnx")
+
+    rewritten_model = ort_rewriter.rewrite(exported_onnx)
```

Comment:
```suggestion
    rewritten_model = optimizer.optimize(exported_onnx)
    rewritten_model = ort_rewriter.rewrite(rewritten_model)
```

`optimize` does the constant folding and dce for the model.

Reply:
After applying this change we get very lean rewritten model

```
===rewritten_model===
number of nodes: 87
  Constant: 34
  Gather: 1
  Slice: 6
  Unsqueeze: 2
  Expand: 1
  Mul: 7
  Sub: 1
  Cast: 3
  Where: 2
  SimplifiedLayerNormalization: 2
  Transpose: 7
  MatMul: 7
  RotaryEmbedding: 2
  Shape: 2
  ConstantOfShape: 1
  GroupQueryAttention: 1
  Add: 3
  Sigmoid: 1
  Pow: 1
  ReduceMean: 1
  Sqrt: 1
  Reciprocal: 1
```